packages/custom-storage/src/sql/SQLStorage.ts

```diff
@@ -143,6 +144,10 @@ export class SQLStorage {
   }
 
   public async deleteVisualization(id: string): Promise<void> {
+    // TODO: Delete dataset if is not used by any other visualization
```

Comment:
I think we can try to add it now, to avoid orphans.

Maybe we can go on something like:
```sql
DELETE 
FROM keplergl_public_v0_datasets 
WHERE id NOT IN 
	(SELECT distinct(dataset) 
     FROM keplergl_public_v0_datasets_vis
    );
```

Adding that as the last query, once the record in the intermediate table has been deleted.